### PR TITLE
Ariel/Unit Test for ExerciseCardViewModel

### PR DESCRIPTION
**Pull request by ArielWangX**
Creating ```ExerciseCardViewModelTest.kt``` and Applying Unit Test on ```ExerciseCardViewModel```.

### CODE REVIEW

File: app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+class ExerciseCardViewModelTest {
+    @get:Rule
+    val navigatorRule = FakeNavigatorRule()
+
+    @get:Rule
+    val coroutineRule = CoroutineRule()
+
+    private val fakeNavigator = navigatorRule.navigator
+    private val fakeExerciseCardRepository = object : ExerciseCardRepository {
+        private var storingExerciseCard = ExerciseCard()
```
Comment (fredliao123) [inline app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt]: nit: storedExerciseCard

File: app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                    ExerciseCardView.State(
+                        notes = ""
+                    ),
+                    awaitItem()
+                )
+            }
+        }
+    }
+
+    // when saved button is clicked, insert exercise card data into database and go back to Exercise screen
```
Comment (fredliao123) [inline app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt]: We can simply use this as the function name

File: app/src/test/java/com/arielwang/workoutlogger/track/TrackViewModelTest.kt [hunk 3]
```diff
@@ -95,8 +92,9 @@ class TrackViewModelTest {
         }
     }
 
+    // check setDefaultTextFieldValue function in weightSection TextFiled
```
Comment (fredliao123) [inline app/src/test/java/com/arielwang/workoutlogger/track/TrackViewModelTest.kt]: I think the function name is good enough to explain these tests. Don't need additional comments here

File: app/src/test/java/com/arielwang/workoutlogger/track/TrackViewModelTest.kt [hunk 4]
```diff
@@ -168,63 +212,175 @@ class TrackViewModelTest {
...
+                            maxChar = 3
+                        )
+                    ),
                     awaitItem()
                 )
             }
         }
     }
 
+    // Check maximum input number of characters in weightSection TextField
```
Comment (fredliao123) [inline app/src/test/java/com/arielwang/workoutlogger/track/TrackViewModelTest.kt]: Better use this as the function name : Check maximum input number of characters in weightSection TextField is 3

**Review by fredliao123 — APPROVED** [review]
Looks good, just a few small comments

File: app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt (hunk 1 above)
Comment (ArielWangX) [reply app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt]: Thank you.

File: app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt (hunk 2 above)
Comment (ArielWangX) [reply app/src/test/java/com/arielwang/workoutlogger/exercisecard/ExerciseCardViewModelTest.kt]: Thank you.

**Pull request merged**